Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Scaffolding/Internal/FbDatabaseModelFactory.cs

```diff
@@ -163,12 +163,18 @@ CASE Coalesce(F.RDB$FIELD_TYPE, 0)
 				 WHEN 0 THEN 'BIGINT'
 				 ELSE 'DECIMAL'
 				END
+                WHEN 23 THEN 'BOOLEAN'
 				WHEN 27 THEN 'DOUBLE PRECISION'
 				WHEN 35 THEN 'TIMESTAMP'
 				WHEN 37 THEN 'VARCHAR(' || (TRUNC(F.RDB$FIELD_LENGTH / CH.RDB$BYTES_PER_CHARACTER)) || ')'
 				WHEN 40 THEN 'CSTRING' || (TRUNC(F.RDB$FIELD_LENGTH / CH.RDB$BYTES_PER_CHARACTER)) || ')'
 				WHEN 45 THEN 'BLOB_ID'
-				WHEN 261 THEN 'BLOB SUB_TYPE ' || F.RDB$FIELD_SUB_TYPE 
+				WHEN 261 THEN 'BLOB SUB_TYPE ' ||
+				  CASE F.RDB$FIELD_SUB_TYPE
+				    WHEN 0 THEN 'BINARY'
+				    WHEN 1 THEN 'TEXT'
+				    ELSE ''
```

Comment:
Although it's probably not happening often, in case it's not `BINARY` or `TEXT` I would return  just the number, instead of empty string.